tests/test_cliCmdImport.py

```diff
@@ -0,0 +1,129 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Unit tests for daf_butler CLI config-dump command.
+"""
+
+import click
+import click.testing
+import unittest
+
+from lsst.daf.butler.cli import butler
+from lsst.daf.butler.cli.utils import clickResultMsg, Mocker, mockEnvVar
+
+
+def makeExpectedKwargs(**kwargs):
+    expected = dict(repo=None,
+                    transfer="auto",
+                    output_run=None,
+                    directory=None,
+                    export_file=None)
+    expected.update(kwargs)
+    return expected
+
+
+class Case(unittest.TestCase):
+
+    def setUp(self):
+        self.runner = click.testing.CliRunner(env=mockEnvVar)
+
+    def run_test(self, inputs, expectedKwargs):
+        """Test command line interaction with import command function.
+
+        Parameters
+        ----------
+        inputs : [`str`]
+            A list of the arguments to the butler command, starting with
+            `import`
+        expectedKwargs : `dict` [`str`, `str`]
+            The expected arguments to the import command function, keys are
+            the argument name and values are the argument value.
+        """
+        result = self.runner.invoke(butler.cli, inputs)
+        self.assertEqual(result.exit_code, 0, clickResultMsg(result))
+        Mocker.mock.assert_called_with(**expectedKwargs)
+
+    def test_minimal(self):
+        """Test only the required parameters, and omit the optional parameters.
+        """
+        expected = makeExpectedKwargs(repo="here", directory="foo", output_run="out")
+        self.run_test(["import", "here",
+                       "foo",
+                       "--output-run", "out"], expected)
+
+    def test_almostAll(self):
+        """Test all the parameters, except export_file which gets its own test
+        case below.
+        """
+        with self.runner.isolated_filesystem():
+            f = open("output.yaml", "w")
```

Comment:
What is the purpose of opening this file? It doesn't seem to be used anywhere.

Reply:
also leftover from addressing a unit test issue, the fix ended up becoming ExportFileCase. Will remove.